azure-mgmt-dns/src/main/java/com/microsoft/azure/management/dns/implementation/MXRecordSetImpl.java

```diff
@@ -57,7 +57,7 @@ protected RecordSetInner prepareForUpdate(RecordSetInner resource) {
                 for (MxRecord recordToRemove : this.recordSetRemoveInfo.mxRecords()) {
                     for (MxRecord record : resource.mxRecords()) {
                         if (record.exchange().equalsIgnoreCase(recordToRemove.exchange())
-                                && (record.preference() == recordToRemove.preference())) {
```

Comment:
Fixing incorrect integer comparison

Reply:
[P3]: If `record.preference()` is null then `record.preference().equals(recordToRemove.preference())` will throw `NullPointerException`. Why P3 because though findbug reported it as an issue, it behave correctly and not a blocker for user.

Work to do: Check for null before attempting to do the comaprison.

Reply:
will put a null check for record.preference()